src/Tokenizer/Analyzer/DataProviderAnalyzer.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer;
+
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\DataProviderAnalysis;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ *
+ * @internal
+ */
+final class DataProviderAnalyzer
+{
+    /**
+     * @return array<DataProviderAnalysis>
+     */
+    public function getDataProviders(Tokens $tokens, int $startIndex, int $endIndex): array
+    {
+        $methods = $this->getMethods($tokens, $startIndex, $endIndex);
+
+        $dataProviders = [];
+        foreach ($methods as $methodIndex) {
+            $docCommentIndex = $tokens->getTokenNotOfKindSibling(
+                $methodIndex,
+                -1,
+                [[T_ABSTRACT], [T_COMMENT], [T_FINAL], [T_FUNCTION], [T_PRIVATE], [T_PROTECTED], [T_PUBLIC], [T_STATIC], [T_WHITESPACE]]
+            );
+            \assert(\is_int($docCommentIndex));
```

Comment:
we don't have alignment about using `assert` in Fixer project. let's align before starting using it.